docs/faq.md

```diff
@@ -38,6 +38,54 @@ targets:
           compiler: dart2js
 ```
 
+## How can I build with multiple configurations
+
+The build system supports two types of builds, "dev" and "release". By default
+with `build_runner` the "dev" version is built by default regardless of the
+command used, build in release mode by passing the `--release` flag. With
+`webdev` the default mode for the `serve` command is dev, and the default mode
+for the `build` command is release. The `build` command can use dev mode with
+the `--no-release` flag.
+
+Options can be configured per mode, and they are merged by key. The `options`
```

Comment:
I don't know what it means that configurations are merged by key. Not necessarily a show stopper - I'm a beginner at this, and this is probably focused on advanced users. Just wanted to point it out.

Reply:
This is intended for anybody to use/read, it is supposed to be the most searchable/easy spot to get common questions answered.

So, if its not meaningful to you we should consider rewording it :)

Reply:
Yeah, I'm trying to decide between expanding on this here, or writing another FAQ or document about option merging and linking it here.

Reply:
We could add a tiny entry like "What is a builder key" and we could link references to "key" to that.

Reply:
Also ya the merging seems like it might be deserving of its own section as it is somewhat complicated to explain.